src/common/buffer.cc

```diff
@@ -984,6 +984,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     char* ptr = _raw->data + _off + _len;
     *ptr = c;
     _len++;
+    assert(_len != 0);
```

Comment:
@branch-predictor 

Yes, this will affect performance. I think assert can be disabled globally by specifing DCMAKE_CXX_FLAGS="-DNDEBUG ..." cmake variable. Any better suggestion?